### PR TITLE
Server Disconnect Popups

### LINKED ISSUE #562: Add Popup Message for Server Unreachable
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description

Currently when trying to join the arena or tavern when the P2P server is offline or the user is having connection issues there is no feedback.

When the server is unreachable, show a popup saying the server is unreachable.

## Intended Design

N/A

## Location

Create a branch from `arena-vertical-slice`

## Related Issues

N/A

## Requirements

N/A

## Additional information

N/A

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

### PR DESCRIPTION
**Pull request by garrettluskey**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the current behaviour?

No information is provided to the user when the server is disconnected.

## What is the new behaviour?

Resolves #562

This adds informative popup messages for users to determine issues in their experience

## Other information

### CODE REVIEW

File: source/Missions/Services/Arena/ArenaTestGameManager.cs [hunk 1]
```diff
@@ -48,7 +48,6 @@ public void StartGame()
             else
             {
                 ScreenManager.PopScreen();
-                // TODO add popup message
```
Comment (EgardA) [inline source/Missions/Services/Arena/ArenaTestGameManager.cs:51]: This is already done

File: source/Missions/Services/Network/LiteNetP2PClient.cs [hunk 2]
```diff
@@ -228,17 +230,15 @@ public void OnPeerConnected(NetPeer peer)
...
             {
                 var peerDisconnectedEvent = new PeerDisconnected(peer, disconnectInfo);
                 messageBroker.Publish(this, peerDisconnectedEvent);
-                return;
             }
-            if (peer == PeerServer)
+            else
             {
-                ServerDisconnected serverDisconnected = new ServerDisconnected();
+                ServerDisconnected serverDisconnected = new ServerDisconnected(peer, disconnectInfo);
```
Comment (EgardA) [inline source/Missions/Services/Network/LiteNetP2PClient.cs]: Sending the peer here seems unneccessary unless you plan on using it later somehow

File: source/Missions/Services/Network/Messages/ServerDisconnected.cs [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
+using LiteNetLib;
 
 namespace Missions.Services.Network.Messages
 {
     public readonly struct ServerDisconnected : IEvent
     {
+        public NetPeer NetPeer { get; }
+        public DisconnectInfo DisconnectInfo { get; }
 
+        public ServerDisconnected(NetPeer netPeer, DisconnectInfo disconnectInfo)
```
Comment (EgardA) [inline source/Missions/Services/Network/Messages/ServerDisconnected.cs]: remove in event too

**Review by EgardA — CHANGES_REQUESTED** [review]

File: source/Missions/Services/Arena/ArenaTestGameManager.cs (hunk 1 above)
Comment (garrettluskey) [reply source/Missions/Services/Arena/ArenaTestGameManager.cs:51]: I don't understand this comment, this line was removed. Could you clarify if I'm misunderstanding

Comment (EgardA) [reply source/Missions/Services/Arena/ArenaTestGameManager.cs:51]: Oops

**Review by EgardA — APPROVED** [review]

**Pull request merged**